storage/client_test.go

```diff
@@ -57,21 +57,20 @@ func createTestStoreWithEngine(t *testing.T, eng engine.Engine, clock *hlc.Clock
 	bootstrap bool, context *storage.StoreContext) (*storage.Store, *util.Stopper) {
 	stopper := util.NewStopper()
 	rpcContext := rpc.NewContext(hlc.NewClock(hlc.UnixNano), security.LoadInsecureTLSConfig(), stopper)
-	var ctx *storage.StoreContext
 	if context == nil {
-		ctx = &storage.TestStoreContext
-	} else {
-		ctx = context
+		// make a copy
+		ctx := storage.TestStoreContext
+		context = &ctx
```

Comment:
This is worth discussion, but not in the scope of this PR (which is fixing some previous mistakes and introducing golint). In the meantime, I've filed https://github.com/barakmich/go-nyet/issues/8